pkg/store/cache/cache.go

```diff
@@ -36,24 +38,24 @@ var (
 // (potentially with a deadline) as in the original user's request.
 type IndexCache interface {
 	// StorePostings stores postings for a single series.
-	StorePostings(blockID ulid.ULID, l labels.Label, v []byte)
+	StorePostings(blockID ulid.ULID, l labels.Label, v []byte, tenant string)
 
 	// FetchMultiPostings fetches multiple postings - each identified by a label -
 	// and returns a map containing cache hits, along with a list of missing keys.
-	FetchMultiPostings(ctx context.Context, blockID ulid.ULID, keys []labels.Label) (hits map[labels.Label][]byte, misses []labels.Label)
+	FetchMultiPostings(ctx context.Context, blockID ulid.ULID, keys []labels.Label, tenant string) (hits map[labels.Label][]byte, misses []labels.Label)
```

Comment:
I've given this some thoughts as well. I agree, using `ctx` would make the function calls cleaner and less verbose. On the other hand, I like putting it explicitly as a variable, as it forces the caller to ensure the tenant information is available, whereas it could be easier to miss when using `ctx`.

In the end I think I prefer the current approach, but happy to change it if there are other opinions :+1:

Reply:
I'm +1 with @jacobbaungard. If the tenant parameter is key to changing the behavior or outcome of the function, I don't think it's a good idea to pass it through context. [The official docs on contexts](https://pkg.go.dev/context) states: `Use context Values only for request-scoped data that transits processes and APIs, not for passing optional parameters to functions.`.

Reply:
Sure, not a blocker. But `tenant` seems to me like `request-scoped data that transits processes and APIs`, which is why I suggested doing that. It also isn't optional anywhere right (we replace with default)?

Reply:
Anything is request scoped data that transits in a system dealing with HTTP requests, no? 😂

Reply:
To me it doesn't seem a good idea to use context as a bag of objects because the change becomes smaller/easier in the codebase. In fact, it hurts discoverability because it hides important parameters from the function's signature.

Reply:
Fair, makes sense.